tests/route/test_route_flap.py

```diff
@@ -441,7 +446,10 @@ def test_route_flap(duthosts, tbinfo, ptfhost, ptfadapter,
     neighbor_type = get_neighbor_info(duthost, dev_port, tbinfo)
     recv_neigh_list = get_all_recv_neigh(duthosts, neighbor_type)
     logger.info("Receiving ports neighbor list : {}".format(recv_neigh_list))
-    ptf_recv_ports = get_all_ptf_recv_ports(duthosts, tbinfo, recv_neigh_list)
+    if 't2' in tbinfo["topo"]["type"] and duthost == duthost_upstream:
+        ptf_recv_ports = get_ptf_recv_ports(duthost, tbinfo)
```

Comment:
what are we doing here ? I could not follow the logic

Reply:
Hi @abdosi,

Earlier before this PR, we are selecting ptf receiving ports from the following:

- Fetch the 'neighbor type' (LeafRouter/SpineRouter/AZNGhub) based on the chosen dev_port from get_dev_port_and_route
- Get all the neighbors of same neighbor type and fetch the ptf receiving ports list based on the retrieved neighbors list.
- Pass the received ptf ports list to send_recv_ping_packet method for further testing.
- **_get_all_ptf_recv_ports_** method is being used for this.

But now with the latest upstream LC changes w.r.t AZNG, instead of selecting receiving ports based on the neighbor type, we are doing the following,
- Check if the topo is '_t2_' and if the selected duthost for this test run is '_upstream_' line card
- If so, get all the ptf receiving ports of the upstream line card and pass them to '_send_recv_ping_packet_'
- For other duts, it will follow the same old way to get the ptf receiving ports list.
- **_get_ptf_recv_ports_** method is being used for this.